R/dichotomize.R

```diff
@@ -37,35 +47,35 @@ setMethod("dichotomize", c("Categories", "numeric"), .dichotomize.categories)
 setMethod("dichotomize", c("Categories", "logical"), .dichotomize.categories)
 #' @rdname dichotomize
 #' @export
-setMethod("dichotomize", c("Categories", "character"), function (x, i) {
-    ind <- names(x) %in% i
+setMethod("dichotomize", c("Categories", "character"), function (categories, selection) {
+    ind <- names(categories) %in% selection
     if (!any(ind)) {
         halt("Category not found") ## make nicer error message
     }
-    return(dichotomize(x, ind))
+    return(dichotomize(categories, ind))
 })
 
 #' @rdname dichotomize
 #' @export
-setMethod("undichotomize", "Categories", function (x) {
-    is.selected(x) <- FALSE
-    return(x)
+setMethod("undichotomize", "Categories", function (categories) {
+    is.selected(categories) <- FALSE
+    return(categories)
 })
 
-.dichotomize.var <- function (x, i) {
-    newcats <- dichotomize(categories(x), i)
-    categories(x) <- newcats
+.dichotomize.var <- function (categories, selection) {
```

Comment:
This is where I don't think `categories` as an argument makes sense: when the method/function is for a variable (or is defined both for `Categories` and `CategoricalArrayVariable`). `categories(categories)`?